### PR TITLE
Add tests for ValidateURLs (kubeadm validation)

### PR DESCRIPTION
**Pull request by prasadkatti**
/sig cluster-lifecycle
/kind feature
/priority backlog

**What this PR does / why we need it**:
Adds unit tests

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by fabriziopandini — APPROVED** [review]
/approve

File: cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go [hunk 1]
```diff
@@ -896,3 +896,50 @@ func TestValidateDiscoveryKubeConfigPath(t *testing.T) {
...
+		{
+			name:           "URL without scheme",
+			urls:           []string{"example.com"},
+			requireHTTPS:   false,
+			expectedErrors: true,
+		},
+	}
+	for _, tc := range tests {
+		actual := ValidateURLs(tc.urls, tc.requireHTTPS, nil)
+		actualErrors := (len(actual) > 0)
```
Comment (neolit123) [inline cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go:940]: non-blocking nit: the extra parentheses are not needed.

**Comment by neolit123** [comment]
@prasadkatti you seem to have a good number of PRs sent for k/k, k/webisite, k/kubeadm. thank you for that!

please note, you can apply for membership as explained [here](https://github.com/kubernetes/community/blob/master/community-membership.md#member)

which grants you the ability to `/lgtm` and `/ok-to-test`.
i can sponsor your membership, but you'd have to find a second sponsor too.

File: cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go (hunk 1 above)
Comment (prasadkatti) [reply cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go]: fixed it.

**Comment by prasadkatti** [comment]
> i can sponsor your membership, but you'd have to find a second sponsor too.

Thanks @neolit123. I would love that. Let me look for a second sponser.

**Comment by fabriziopandini** [comment]
/lgtm
Looking forward to see more contribution from @prasadkatti

**Pull request merged**